### DISCUSSION TITLE
JGroups 5.4.0 no longer has LockService

### DISCUSSION THREAD

**Discussion by davsclaus** [belaban/JGroups#870] [General]
Upgrading from 5.3.13 to 5.4.0 then we have failing code compilation because org.jgroups.blocks.locking.LockService is no longer included in the JAR.

Is this completely removed or is there something else that can be used for cluster locking?

**Comment by rhusar** [comment]
For proper clustered locks you could use Infinispan - https://infinispan.org/docs/stable/titles/embedding/embedding.html#clustered_locks

**Reply by tristantarrant** [reply]
Actually, don't use clustered locks at all, as Bela points out below.

**Comment by belaban** [comment]
Yes, clustered locks were removed in 5.4, see https://belaban.blogspot.com/2020/11/i-hate-distributed-locks.html?m=1 for reasons.Von meinem/meiner Galaxy gesendet
-------- Ursprüngliche Nachricht --------Von: Claus Ibsen ***@***.***> Datum: 04.12.24  12:07  (GMT+01:00) An: belaban/JGroups ***@***.***> Cc: Subscribed ***@***.***> Betreff: [belaban/JGroups] JGroups 5.4.0 no longer has LockService (Discussion #870) 
Upgrading from 5.3.13 to 5.4.0 then we have failing code compilation because org.jgroups.blocks.locking.LockService is no longer included in the JAR.
Is this completely removed or is there something else that can be used for cluster locking?

**Comment by davsclaus** [comment]
Thanks for the help. We will remove the lock in camel-jgroups.
